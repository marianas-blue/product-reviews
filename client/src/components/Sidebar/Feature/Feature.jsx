/* eslint-disable react/jsx-one-expression-per-line */
/* eslint-disable no-useless-constructor */
/* eslint-disable react/prefer-stateless-function */
import React from 'react';
import { Descriptor, Border, Title, Rating, Keep } from './Feature.style';

class Feature extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <Border>
        <Title>By feature</Title>
        <Descriptor>
          Sound quality
          <Keep>
            <img src="https://s3.amazonaws.com/product-reviews-hr110/Icons/stars.png" alt="stars" height="25px" />
            <Rating>4.4</Rating>
          </Keep>
        </Descriptor>
        <Descriptor>
          Battery life
          <Keep>
            <img src="https://s3.amazonaws.com/product-reviews-hr110/Icons/stars.png" alt="stars" height="25px" />
            <Rating>4.1</Rating>
          </Keep>
        </Descriptor>
        <Descriptor>
          Noise cancellation
          <Keep>
            <img src="https://s3.amazonaws.com/product-reviews-hr110/Icons/stars.png" alt="stars" height="25px" />
            <Rating>4.1</Rating>
          </Keep>
        </Descriptor>
      </Border>
    );
  }
}

export default Feature;
