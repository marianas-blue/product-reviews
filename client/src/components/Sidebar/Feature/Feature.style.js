import styled from 'styled-components';

export const Descriptor = styled.div`
font-size: 14px;
display: flex;
justify-content: space-between;
align-items: center;
`;

export const Border = styled.div`
border-top: 1px solid #e7e7e7;
font-size 14px;
padding: 15px 15px 15px 0px
`;

export const Title = styled.button`
font-weight: bolder;
font-size: 17px;
line-height: 1.255;
border: none; 
padding: 0;
background-color: transparent;
outline: none;
text-align: left;
`;

export const Rating = styled.div`
  color: #797979;
  display: inline-block;
`;

export const Keep = styled.div`
display: flex;
align-items: center;
margin-top: 7px;
margin-bottom: 7px;
`;


