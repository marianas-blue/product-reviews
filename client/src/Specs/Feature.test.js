/* eslint-disable react/jsx-filename-extension */
/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';

import Feature from '../components/Sidebar/Feature/Feature';

describe('Feature', () => {
  it('should render correctly', () => {
    const component = shallow(<Feature />);

    expect(component).toMatchSnapshot();
  });
});
