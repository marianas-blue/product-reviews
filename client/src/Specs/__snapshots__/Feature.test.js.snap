// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature should render correctly 1`] = `
<styled.div>
  <styled.button>
    By feature
  </styled.button>
  <styled.div>
    Sound quality
    <styled.div>
      <img
        alt="stars"
        height="25px"
        src="https://s3.amazonaws.com/product-reviews-hr110/Icons/stars.png"
      />
      <styled.div>
        4.4
      </styled.div>
    </styled.div>
  </styled.div>
  <styled.div>
    Battery life
    <styled.div>
      <img
        alt="stars"
        height="25px"
        src="https://s3.amazonaws.com/product-reviews-hr110/Icons/stars.png"
      />
      <styled.div>
        4.1
      </styled.div>
    </styled.div>
  </styled.div>
  <styled.div>
    Noise cancellation
    <styled.div>
      <img
        alt="stars"
        height="25px"
        src="https://s3.amazonaws.com/product-reviews-hr110/Icons/stars.png"
      />
      <styled.div>
        4.1
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;
